.github/workflows/push-pull.yml

```diff
@@ -43,52 +44,58 @@ jobs:
       # testing and preventing merging).
       CTEST_NO_WARNINGS_ALLOWED: true
 
-      # On some platforms, this is used to select a different toolset.
-      # For example, uncomment the line below on Centos 7 to use a
-      # newer GCC (assumes the proper toolset packages are installed).
-      #LAUNCHER: scl enable devtoolset-7 --
-
-      # Set the build type (this currently has no effect).
-      # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html
-      # CMAKE_BUILD_TYPE: Release
-
-      # Set the cmake generator (build system) to use.
-      # See https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html
-      # CMAKE_GENERATOR: Ninja
       CMAKE_GENERATOR: Unix Makefiles
     steps:
       - name: Checkout Source
         uses: actions/checkout@v2
         id: git
-      - name: Filesystem Prep
-        run: mkdir -p /sccache
-        id: fs
+      - name: Create keys
+        id: keys
+        run: |
+          echo "::set-output name=key1::$(( ${{ github.run_number }} - 1))"
+          echo "::set-output name=key2::$(( ${{ github.run_number }} - 2))"
+          echo "::set-output name=key3::$(( ${{ github.run_number }} - 3))"
+          echo "::set-output name=key4::$(( ${{ github.run_number }} - 4))"
+          echo "::set-output name=key5::$(( ${{ github.run_number }} - 5))"
       - name: Populate sccache
         uses: actions/cache@v2
         env:
           cache-name: sccache
         id: cache
         with:
-          path: /sccache
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ github.ref }}-${{ hashFiles('include/ibamr/config.h','ibtk/include/ibtk/config.h.in') }}
+          path: ~/.cache/sccache
+          # A fundamental problem with the cache is that it is identified by a
+          # single key and, should a cache with that key already exist, the
+          # cache will not be updated. This makes sense for things like NPM but
+          # not so much for sccache, since we get a new and unique cache each
+          # time we rebuild the library with a new branch.
+          #
+          # To get around this, always make the key unique and rely on
+          # restore-keys to find the highest-indexed key available. Explicitly
+          # try to restore things from the most recent builds.
```

Comment:
This is trying to avoid recompiling stuff when possible?

How do you know the key corresponds to the current build?

Any reason for 5 keys?

Reply:
1. Yes.

2. It won't. The key I pick depends on the build number in GitHub's CI, which should be new. 

3. Its a wild guess. The point here is that I try to find the most recent of the last five builds to look for cached compiler results. I wanted more than one so that if we start, say, 4 builds at once, then the last one we start will still try to find something reasonable.

This is a really hacky part of the CI (which is probably why it didn't work before). The intention of this GitHub action is for storing dependencies, not build artifacts, so we have to provide some extra information to make it do what we want (find a recently compiled copy of IBAMR and try to reuse as many object files as possible).